python-threatexchange/threatexchange/dataset.py

```diff
@@ -76,19 +80,47 @@ def get_fetch_checkpoint(self) -> FetchCheckpoint:
             return FetchCheckpoint(0, 0)
         return FetchCheckpoint.deserialize(checkpoint.read_text())
 
-    def get_indicator_checkpoint(self) -> int:
-        checkpoint = self._fetch_checkpoint_path(self.INDICATOR_SUFFIX)
+    def get_indicator_checkpoint(self, privacy_group) -> t.Dict:
+        values = {
+            "last_stop_time": 0,
+            "last_run_time": 0,
+            "threat_types": None,
+            "url": None,
+        }
+        checkpoint = self._indicator_checkpoint_path(privacy_group)
         if not checkpoint.exists():
-            return 0
-        return int(checkpoint.read_text())
+            return values
+        with checkpoint.open("r+") as f:
+            values["last_stop_time"] = int(f.readline().strip())
+            values["last_run_time"] = int(f.readline().strip())
+            values["threat_types"] = f.readline().strip()
+            values["threat_types"] = (
+                values["threat_types"].split(" ")
+                if values["threat_types"] != ""
+                else None
+            )
+            values["url"] = f.readline().strip()
+            values["url"] = values["url"] if values["url"] != "" else None
+        return values
```

Comment:
(maybe blocking): This style of read/writing files is super fragile, and it's easy to make mistakes or make backwards incompatible changes.

Please instead switch to reading and writing json dicts, which are easier to create forwards/backwards compabile behavior.